### ISSUE TITLE
Cannot start the example app.

### ISSUE THREAD

**Issue by buithugiang** [christabor/flask_jsondash#82]
When I run the example app both with method 1 and method 3, I got this trouble of viewing the chart, I don't know if I miss any configuration, but it is like loading forever but not showing the chart.
I have attached the image, would you pls take a look!
<img width="1001" alt="screen shot 2017-01-25 at 8 37 53 am" src="https://cloud.githubusercontent.com/assets/4782134/22274518/8fcc1eb8-e2d9-11e6-91cd-a113e4a5cb96.png">

<img width="1018" alt="screen shot 2017-01-25 at 8 38 11 am" src="https://cloud.githubusercontent.com/assets/4782134/22274581/eeb16082-e2d9-11e6-9521-5a886ae5a846.png">

**Comment by christabor** [comment]
I'll take a look today!

**Comment by christabor** [comment]
My gut feeling is that the data source config might be wrong simply because the dashboard is loading with the chart correctly, but the visualization is not working. Can you post the json for the config as well as the data source payload?

**Comment by christabor** [comment]
Looks like (from the screenshot) you're using the plotly chart type, is that correct? Does your endpoint use a plotly compatible config? Also I can't recall, but you may need to check the custom override option there.

**Comment by buithugiang** [comment]
Thank you for taking a look!

<img width="849" alt="screen shot 2017-01-26 at 5 29 32 pm" src="https://cloud.githubusercontent.com/assets/4782134/22328079/f1780a46-e3ec-11e6-877c-c12e9b85cd23.png">
 I've tried your example and got the config:
{
    "created_by": "anonymous",
    "modules": [],
    "name": "Testing",
    "id": "f2d19d36-e3b1-11e6-82a7-f45c89aaf953",
    "date": "2017-01-26 17:26:50.392000",
    "username": "anonymous"
}
Here is the payload:

{"bar2": [11, 18, 11, 48, 9], "bar1": [1, 2, 30, 12, 100], "bar3": [44, 71, 85, 98, 20]}

**Comment by christabor** [comment]
Hey, I noticed that you didn't have any charts specific in your config. Did you happen to save and reload the page? Here is what I used and it is loading for me:

```json
{
    "username": "anonymous",
    "name": "Testing",
    "modules": [
        {
            "name": "Testing",
            "family": "C3",
            "refresh": false,
            "height": "400",
            "width": "500",
            "dataSource": "http://127.0.0.1:5004/bar",
            "override": false,
            "refreshInterval": "",
            "guid": "56259aa4-c28f-5279-80b0-97de1da5d69f",
            "type": "bar",
            "order": ""
        }
    ],
    "created_by": "anonymous",
    "date": "2017-01-26 12:24:10.546491",
    "id": "1af6cf33-e405-11e6-9010-a0999b0aebbf"
}
```

Similar payload, as it's serving from the same endpoint. I wonder if it's having trouble loading the preview, but works after you save it?

**Comment by buithugiang** [comment]
Hi,

<img width="1280" alt="screen shot 2017-01-30 at 11 37 30 am" src="https://cloud.githubusercontent.com/assets/4782134/22412387/7b8c476a-e6e0-11e6-9525-c45b76456f60.png">

I noticed that when I clicked on "Save Module" button, no request was made, the dialog just disappeared, and the loader just like running forever.

<img width="1276" alt="screen shot 2017-01-30 at 11 37 41 am" src="https://cloud.githubusercontent.com/assets/4782134/22412435/d18836ce-e6e0-11e6-8c11-aae9cf39f210.png">

When I manually added the config that you posted here and saved it, it worked.

**Comment by christabor** [comment]
Ok, I'll look into why the preview is not loading. For now, saving after adding a chart should get you going. Thanks!

**Comment by buithugiang** [comment]
Sure, thanks a lot and looking forward to your investigation.

**Comment by christabor** [comment]
I believe this is fixed. I realize that the "save module" text is misleading now, as it kind of implies that the chart will be saved onClick. I have forced it to save when this added, which fixes this problem, and reduces the need to click a second time. Also reduces a small amount of js code as it is no longer required to add an initial preview. I've added a new version to pypi as well as a new tag `5.0.1`.

Thanks for checking this out, much appreciated!
